helpers/isChatToYourself.js

```diff
@@ -0,0 +1,28 @@
+const {User} = require('../databases/models');
+const UsersFunction = require('../databases/functions/users');
+
+module.exports = {
+  isChatToYourself: async (my_user_id, member) => {
+    let user = await UsersFunction.findUserById(User, member);
+    if (!user) {
+      return {
+        success: false,
+        message: 'User not found'
+      };
+    }
+    const user_ids = new Set([user.user_id]);
+    if (!user.is_anonymous) {
+      const anon_user = await UsersFunction.findAnonymousUserId(User, user.user_id);
+      user_ids.add(anon_user.user_id);
+    } else {
+      const sign_user = await UsersFunction.findSignedUserId(User, user.user_id);
+      user_ids.add(sign_user.user_id);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is making unnecessary database calls to fetch the user and their anonymous/signed-in counterpart. This could be optimized by passing the `user_id` directly instead of fetching it from the database. Also, the use of a Set for storing just one or two user ids seems overkill. A simple comparison would suffice.

```diff
-    let user = await UsersFunction.findUserById(User, member);
-    if (!user) {
-      return {
-        success: false,
-        message: 'User not found'
-      };
-    }
-    const user_ids = new Set([user.user_id]);
-    if (!user.is_anonymous) {
-      const anon_user = await UsersFunction.findAnonymousUserId(User, user.user_id);
-      user_ids.add(anon_user.user_id);
-    } else {
-      const sign_user = await UsersFunction.findSignedUserId(User, user.user_id);
-      user_ids.add(sign_user.user_id);
-    }

+    const isChattingToSelf = (my_user_id === member || my_user_id === UsersFunction.getCounterpartUserId(member));
```

Please note that you need to implement `getCounterpartUserId` function in `UsersFunction` which should return the counterpart user id for a given user id.



<!-- This is an auto-generated comment by OSS CodeRabbit -->